src/index.ts

```diff
@@ -0,0 +1,315 @@
+import { version } from '../package.json';
+
+type MandatoryField = {
+  name: string;
+  optional?: false;
+};
+type OptionalField = {
+  name: string;
+  optional: true;
+};
+type Field = MandatoryField | OptionalField;
+
+type DateField = {
+  type: 'date';
+} & Field;
+
+type DateTimeField = {
+  type: 'datetime';
+} & Field;
+
+type DurationField = {
+  type: 'duration';
+  time_unit: 'milliseconds' | 'seconds' | 'minutes' | 'hours';
+} & Field;
+
+type CurrencyCode =
```

Comment:
Where did you get this list from?

Reply:
https://en.wikipedia.org/wiki/ISO_4217#Active_codes

Reply:
Ah, I see. We need to make sure it's the same list that Iris uses for validation. You can find it here (though for some reason it's not in alphabetical order): https://github.com/geckoboard/iris/blob/f4caaed4e3f9cfed43c1a19cb4e7b5fe19eea947/iso4217/codes.go

Reply:
OK. Fwiw this list is linked in the docs.

Reply:
It would be interesting to know what the difference is because I counted one extra currency in this list compared to Iris's.

Reply:
We seem to have a few former currency in Iris it seems :
```
HRK
MRO
STD
CUC
VEF
BYR
```
and newer ones from the wikipedia:
```
BYN
MRU
SLE
STN
UYW
VED
VES
```

Reply:
Should we in fact update Iris rather than use outdated codes?